cooked-validators/src/Cooked/MockChain/Monad/Direct.hs

```diff
@@ -239,17 +252,56 @@ instance (Monad m) => MonadMockChain (MockChainT m) where
 
   slotConfig = asks mceSlotConfig
 
--- | Check 'validateTx' for details
-validateTx' :: (Monad m) => Pl.Tx -> MockChainT m Pl.TxId
-validateTx' tx = do
+-- | This validates a given 'Pl.Tx' in its proper context; this is a very tricky thing to do. We're basing
+--  ourselves off from how /plutus-apps/ is doing it.
+--
+--  TL;DR: we need to use "Ledger.Index" to compute the new 'Pl.UtxoIndex', but we neet to
+--  rely on "Ledger.Validation" to run the validation akin to how it happens on-chain, with
+--  proper checks on transactions fees and signatures.
+--
+--  For more details, check the following relevant pointers:
+--
+--  1. https://github.com/tweag/plutus-libs/issues/92
+--  2. https://github.com/input-output-hk/plutus-apps/blob/03ba6b7e8b9371adf352ffd53df8170633b6dffa/plutus-ledger/src/Ledger/Tx.hs#L126
+--  3. https://github.com/input-output-hk/plutus-apps/blob/03ba6b7e8b9371adf352ffd53df8170633b6dffa/plutus-contract/src/Wallet/Emulator/Chain.hs#L209
+--  4. https://github.com/input-output-hk/plutus-apps/blob/03ba6b7e8b9371adf352ffd53df8170633b6dffa/plutus-contract/src/Wallet/Emulator/Wallet.hs#L314
+--
+-- Finally; because 'Pl.fromPlutusTx' doesn't preserve signatures, we need the list of signers
+-- around to re-sign the transaction.
+runTransactionValidation ::
+  Pl.Slot ->
+  Pl.SlotConfig ->
+  Pl.UtxoIndex ->
+  -- | List of required signers
+  [Pl.PaymentPubKeyHash] ->
+  -- | List of signers
+  [Wallet] ->
+  Pl.Tx ->
+  (Pl.UtxoIndex, Maybe Pl.ValidationErrorInPhase, [Pl.ScriptValidationEvent])
+runTransactionValidation s slotCfg ix reqSigners signers tx =
+  let ((e1, idx'), evs) = Pl.runValidation (Pl.validateTransaction s tx) (Pl.ValidationCtx ix slotCfg)
+      cIx = either (error . show) id $ Pl.fromPlutusIndex ix
+      cTx = either (error . show) id $ Pl.fromPlutusTx cIx reqSigners tx
```

Comment:
Just double-checking `error` is the right choice here instead of reporting these as `MockChainError`.

Also, this is somewhat nitpicky, but `cIx` and `cTx` look similar enough on a quick glance (at least, it took me a couple of seconds to tell them apart). Perhaps worth renaming `cIx` to `cIdx` or something like that?

Reply:
Because this is an unrecoverable error, I think `error` is fine here, helps keep the function pure. I'm onboard with lifting it to `MockChainError`s after we understand what kinds of errors can we actually get from here